materialize-mysql/dynamic_varchar.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	stdsql "database/sql"
+)
+
+type colDetails struct {
+	maxLength int
+}
+
+// getVarcharDetails queries the database for VARCHAR type columns to determine their actual maximum
+// lengths and character set. The result map has keys that are table identifiers, and each one of those has a value
+// that is a map of column identifiers to the details of the column. Since Redshift always
+// converts all letters to lowercase for table or column identifiers, the returned identifiers are
+// also all lowercase.
+func getVarcharDetails(ctx context.Context, dbName string, conn *stdsql.Conn) (map[string]map[string]colDetails, error) {
```

Comment:
@williamhbaker in case of MySQL, since there is no notion of a `schema` in a database (in mysql, the words `schema` and `database` refer to the same thing), I'm marking database as required 👍🏽